metadata/write_jsons.py

```diff
@@ -0,0 +1,256 @@
+# output params specified in proc_mpi_sl.template under each fault dir
+# TODO auto retrive from the templates rather than hardcode
+BB_KEYS = ['cores', 'run_time', 'fd_count', 'dt', 'start_time', 'end_time']
+HF_KEYS = ['cores', 'run_time', 'fd_count', 'nt', 'nsub_stoch', 'start_time', 'end_time']
+LF_KEYS = ['cores', 'run_time', 'nt', 'nx', 'ny', 'nz', 'start_time', 'end_time']
+IM_KEYS = ['start_time', 'end_time']
+POST_EMOD_KEYS = ['start_time', 'end_time']
+KEY_DICT = {'BB': BB_KEYS, 'HF': HF_KEYS, 'LF': LF_KEYS, 'IM_calc': IM_KEYS, 'post_emod': POST_EMOD_KEYS}
+
+CH_LOG = 'ch_log'
+JSON_DIR = 'jsons'
+ALL_META_JSON = 'all_sims.json'
+PARAMS_BASE = 'params_base.py'
+
+
+def get_all_sims_dict(fault_dir):
+    """
+    :param fault_dir: abs path to a single fault dir
+    :return: all_sims_dict or None
+    """
+    # Get all ch_log directories
+    hh = get_hh(fault_dir)
+    result_dict = {}
+
+    # Find all ch_log and Rlog directories (to allow handling of the different
+    # folder structures
+    ch_log_dirs = glob.glob(os.path.join(fault_dir, "**", "ch_log"))
+    rlog_dirs = glob.glob(os.path.join(fault_dir, "**", "LF", "**", "Rlog"),
+                          recursive=True)
+    if len(ch_log_dirs) == 0:
+        print("No ch_log directories found for {}".format(fault_dir))
+        return
+
+    for ch_log_dir in ch_log_dirs:
+        # Iterate over the log files
+        for f in os.listdir(ch_log_dir):
+            segs = f.split('.')
+            sim_type = segs[0]
+            # exclude submit time log file (have 3 segs)
+            if len(segs) == 4 and sim_type in KEY_DICT.keys():
+                with open(os.path.join(ch_log_dir, f), 'r') as log_file:
+                    buf = log_file.readlines()
+
+                raw_data = buf[0].strip().split()
+                realization = raw_data[0]
+                data = raw_data[2:]
+
+                # Realization entry already exists
+                if realization in result_dict.keys():
+                    result_dict[realization].update({sim_type: {}})
+                # Create the realization entry
+                else:
+                    result_dict[realization] = {'common': {'hh': hh},
+                                                  sim_type: {}}
+
+                # Retrieve the metadata
+                for i in range(len(data)):
+                    key = KEY_DICT[sim_type][i]
+                    if key == 'run_time':
+                        data[i] = "{:.5f} hour".format(float(data[i]))
+                    result_dict[realization][sim_type][key] = data[i]
+
+                # Extra LF metadata
+                if sim_type == 'LF' and \
+                        result_dict[realization][sim_type].get('total_memory_usage') is None:
+                    # Get the rlog dir for the current realization
+                    cur_rlog_dir = [rlog_dir for rlog_dir in rlog_dirs if
+                                    realization in rlog_dir.split("/")][0]
```

Comment:
Any chance that this list is empty, and runs into IndexError?

Reply:
Yeah, good catch. Will fix.